### PR TITLE
54 integrate login/registration with backend

### LINKED ISSUE #54: Integrate login/registration with backend
- [x] Registering creates an account on backend
- [x] Login is based on accounts stored in database
- [ ] ~~Creating a shop creates it in the database~~ (waiting on a single endpoint for address + shop creation)

### PR DESCRIPTION
**Pull request by HKanwal**
Closes #54

### CODE REVIEW

File: src/next-app/pages/create-account.tsx [hunk 1]
```diff
@@ -1,84 +1,132 @@
...
+              const setterMap = new Map<string, Dispatch<SetStateAction<Field>>>();
+              setterMap.set('email', setEmail);
+              setterMap.set('username', setUsername);
+              setterMap.set('password', setPassword);
+              for (let key of Object.keys(response)) {
+                const setter = setterMap.get(key);
+                setter &&
+                  setter((prev) => {
+                    return { ...prev, errors: response[key] };
+                  });
```
Comment (aggarwal-h) [inline src/next-app/pages/create-account.tsx:64]: Errors caught here are not being displayed to the user. For example, a duplicate username results in an error 400 but it's not being shown anywhere in the UI.

File: src/django-app/django-project/config/settings.py [hunk 2]
```diff
@@ -26,8 +26,10 @@
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
 
-ALLOWED_HOSTS = []
+ALLOWED_HOSTS = ["*"]
 
+# SECURITY WARNING: don't allow all CORS origins in production
+CORS_ALLOW_ALL_ORIGINS = True
```
Comment (aggarwal-h) [inline src/django-app/django-project/config/settings.py:32]: Let's do something like:
```python3
if DEBUG:
     CORS_ALLOW_ALL_ORIGINS = True
```

File: src/next-app/pages/invite.tsx [hunk 3]
```diff
@@ -7,7 +7,7 @@ import styles from '../styles/pages/Invite.module.css';
 const InvitePage: NextPage = () => {
   return (
     <div className={styles.container}>
-      <Header goBackToHref="/create-account" title="Invite Employees" />
+      <Header goBackToHref="/create-shop" title="Invite Employees" />
```
Comment (aggarwal-h) [inline src/next-app/pages/invite.tsx:10]: Why do we need this `goBackToHref`? Can't we do something like `router.back()`?

There is also an issue with this:
When you login as ShopOwner, then use the hamburger menu to select "Employees", you get taken to the Invite Employee Page. Then, if you use this back button, you get taken to "Create Shop". If you click back one more time, you get taken to "Create Account" page.

**Review by aggarwal-h — CHANGES_REQUESTED** [review]
Hey @HKanwal, overall this looks good. I have listed some suggestions below.

File: src/next-app/pages/create-account.tsx (hunk 1 above)
Comment (HKanwal) [reply src/next-app/pages/create-account.tsx]: Resolved. Updated logic for new api format.

File: src/django-app/django-project/config/settings.py (hunk 2 above)
Comment (HKanwal) [reply src/django-app/django-project/config/settings.py]: Resolved

File: src/next-app/pages/invite.tsx (hunk 3 above)
Comment (HKanwal) [reply src/next-app/pages/invite.tsx]: Resolved. Changed to `router.back()` as recommended.

**Comment by HKanwal** [comment]
Thanks for the thorough review @aggarwal-h ! Can you please review that the mentioned issues have been resolved?

**Review by aggarwal-h — APPROVED** [review]
Looks good now @HKanwal.

One tiny issue I noticed: Using the same email for registration results in an error but no error is shown to the user. We can fix this later on. Other than that, it's good to go for the POC demo!

**Pull request merged**